inputremapper/gui/reader_service.py

```diff
@@ -267,68 +268,85 @@ def _create_event_pipeline(self, sources: List[evdev.InputDevice]) -> ContextDum
         context = ContextDummy()
         # create a context for each source
         for device in sources:
+            device_hash = get_device_hash(device)
             capabilities = device.capabilities(absinfo=False)
 
             for ev_code in capabilities.get(EV_KEY) or ():
-                context.notify_callbacks[(EV_KEY, ev_code)].append(
+                input_config = InputConfig(
+                    type=EV_KEY, code=ev_code, origin=device_hash
+                )
```

Comment:
I think the constructor should hash the device instead

```py
input_config = InputConfig(
    type=EV_KEY, code=ev_code, origin=device
)
```

Reply:
Not possible. We need to construct the InputConfig from the json file. 

Also it would make evdev.InputDevice a dependency for InputConfig, which is not really necessary.

Reply:
Ok, should `origin` maybe be renamed to `origin_hash` then?

In any case, there should be a comment above the member variable explaining that this is a hash

Reply:
Maybe use a type alias like "DeviceHash = str", and then change

```py
def get_device_hash(device: evdev.InputDevice) -> str:
```

to

```py
def get_device_hash(device: evdev.InputDevice) -> DeviceHash
```

and

```py
origin: Optional[str] = None
```

to 

```py
origin: Optional[DeviceHash] = None
```